odc/stats/plugins/l34_utils/l4_water.py

```diff
@@ -3,45 +3,69 @@
 NODATA = 255
 
 
-def water_classification(xx, intertidal_mask, water_persistence):
+def water_classification(xx, water_persistence):
 
+    # Replace nan with nodata
     l4 = expr_eval(
-        "where(((a==223)|(a==221))&(b==1), 101, a)",
-        {"a": xx.classes_l3_l4.data, "b": water_persistence},
+        "where((a==a), a, nodata)",
+        {"a": xx.level_3_4.data},
         name="mark_water",
         dtype="uint8",
+        **{"nodata": NODATA},
+    )
+
+    intertidal_mask = expr_eval(
+        "where(a==_u, 1, 0)",
+        {"a": l4},
+        name="mask_intertidal",
+        dtype="uint8",
+        **{"_u": 223},
```

Comment:
there is no need for this block, can be done in the expression line 29 and 37

Reply:
Removed the intertidal mask.